pkg/token/token.go

```diff
@@ -210,23 +266,56 @@ func (g generator) GetWithRoleForSession(clusterID string, roleARN string, sess
 		stsAPI = sts.New(sess, &aws.Config{Credentials: creds})
 	}
 
-	return g.GetWithSTS(clusterID, stsAPI)
+	return g.GetWithSTS(clusterID, roleARN, stsAPI)
```

Comment:
At that stage, the roleARN is only used to determine the filename for the cache file. If it's empty, the corresponding filename segment will be empty as well, which will produce expected behaviour. That is, if a roleARN is specified, only a cache with that roleARN will be used, if not, only a cache without a roleARN will be used.

Reply:
Makes sense. Although, someone could use multiple IAM users without a role, on the same cluster, and then get the same token, correct?

Reply:
If someone changes their credentials without assuming a role with the same clusterID, they'll get the cache for the former, yes. I'm not entirely sure how this could be fixed easily, since you'd need to know before making any calls to AWS what your calling identity is if you want to avoid that.